aw/Models/AWApi/AWReach.swift

```diff
@@ -0,0 +1,140 @@
+import MapKit
+import Foundation
+import UIKit
+
+struct AWReach: Codable {
+    let difficulty: String
+    let condition: String
+    //swiftlint:disable:next identifier_name
+    let id: Int
+    let name: String
+    let putInLat: String?
+    let putInLon: String?
+    let lastGageReading: String?
+    let section: String
+    let unit: String?
+    let takeOutLat: String?
+    let takeOutLon: String?
+    let state: String?
+    let delta: String?
+    //swiftlint:disable:next identifier_name
+    let rc: String?
+
+    enum CodingKeys: String, CodingKey {
+        //swiftlint:disable:next identifier_name
+        case id, name, section, unit, state, delta, rc
+        case difficulty = "class"
+        case condition = "cond"
+        case putInLat = "plat"
+        case putInLon = "plon"
+        case lastGageReading = "last_gauge_reading"
+        case takeOutLat = "tlat"
+        case takeOutLon = "tlon"
+    }
+
+    func distanceFrom(location: CLLocation) -> Double? {
+        guard let lat = putInLat, let latitude = Double(lat),
+            let lon = putInLon, let longitude = Double(lon) else { return nil }
+
+        let reachCoordinate = CLLocation(latitude: latitude, longitude: longitude)
+
+        guard CLLocationCoordinate2DIsValid(reachCoordinate.coordinate) else { return nil }
+
+        return reachCoordinate.distance(from: location)
+    }
+}
+
+struct AWReachInfo: Codable {
+    //swiftlint:disable:next identifier_name
+    let id: Int
+    let abstract: String?
+    let avgGradient: Int16?
+    let photoId: Int32?
+    let length: String?
+    let maxGradient: Int16?
+    let description: String?
+    let shuttleDetails: String?
+    let zipcode: String?
+
+    enum CodingKeys: String, CodingKey {
+        //swiftlint:disable:next identifier_name
+        case id, abstract, length, description, zipcode
+        case avgGradient = "avggradient"
+        case photoId = "photoid"
+        case maxGradient = "maxgradient"
+        case shuttleDetails = "shuttledetails"
+    }
+}
+
+struct AWReachGage: Codable {
+    let gageID: Int
+    let gageReading: String?
+    let rangeComment: String?
+
+    enum CodingKeys: String, CodingKey {
+        case gageID = "gauge_id"
+        case gageReading = "gauge_reading"
+        case rangeComment = "range_comment"
+    }
+}
+
+struct AWRapid: Codable {
+    let name: String?
+    let description: String?
+    let rapidid: Int
+    let difficulty: String?
+    let rlat: String?
+    let rlon: String?
+}
+
+struct AWReachMain: Codable {
+    let info: AWReachInfo
+}
+
+struct AWReachDetailResponse: Codable {
```

Comment:
I thought you could use dot notation in Swift instead of creating one class per level? Might be misremembering.

Reply:
We can do nested classes which will be a little cleaner. I did that a little further down with [`AWReachDetailResponse.Sub.Main`](https://github.com/abkfenris/aw-ios/blob/0eb5d491df2b25c8f6dd082727a21b070acd0afb/aw/Models/AWApi/AWReach.swift#L132) as that is less likely to be referenced directly as `AWReachDetailResponse` or `AWReachDetail`. Once you decode JSON into a set of nested Codables, you can use dot notation. [`callback(detail.view.main)`](https://github.com/abkfenris/aw-ios/blob/0eb5d491df2b25c8f6dd082727a21b070acd0afb/aw/Helpers/AWApiHelper.swift#L224) is where the `AWReachDetailResponse` gets used, and the important bits returned.